src/test/java/org/jenkins/plugins/lockableresources/LockStepTest.java

```diff
@@ -513,4 +515,89 @@ public void evaluate() throws Throwable {
 			}
 		});
 	}
+
+	@Issue("JENKINS-40879")
+	@Test
+	public void parallelLockRelease() throws Exception {
+		story.addStep(new Statement() {
+			@Override
+			public void evaluate() throws Throwable {
+				LockableResourcesManager.get().createResource("resource1");
+				LockableResourcesManager.get().createResource("resource2");
+				LockableResourcesManager.get().createResource("resource3");
+				LockableResourcesManager.get().createResource("resource4");
+				WorkflowJob j = story.j.jenkins.createProject(WorkflowJob.class, "j");
+				j.setDefinition(new CpsFlowDefinition(
+						"lock(resource: 'resource1') {\n" +
+								"    sleep 1\n" +
```

Comment:
Not to mention that the behavior we're testing for is itself kinda flaky - you need a fair number of builds and at least two (more to be safe!) `lock`s to get it to happen consistently.